storage/store.go

```diff
@@ -1430,20 +1430,72 @@ func (s *Store) WaitForRangeScanCompletion() int64 {
 // updateStoreStatus updates the store's status proto.
 func (s *Store) updateStoreStatus() {
 	now := s.ctx.Clock.Now().WallTime
+	timestamp := proto.Timestamp{WallTime: now}
 	scannerStats := s.scanner.Stats()
 
+	// Get the zone configs.
+	zoneMap, err := s.Gossip().GetInfo(gossip.KeyConfigZone)
+	if err != nil || zoneMap == nil {
+		log.Error("unable to get zone configs")
+		return
+	}
+
+	// Get the leader count and replication count.
+	// TODO(bram): Consider moving this to be part of the range scanner directly.
+	var leaderRangeCount, replicatedRangeCount, availableRangeCount int32
+	s.mu.Lock()
+	for raftID, rng := range s.ranges {
+		zoneConfig := zoneMap.(PrefixConfigMap).MatchByPrefix(rng.Desc().StartKey).Config.(*proto.ZoneConfig)
+		raftStatus := s.RaftStatus(raftID)
+		if raftStatus == nil {
+			continue
+		}
+		if raftStatus.SoftState.RaftState == raft.StateLeader {
+			leaderRangeCount++
+			// TODO(bram): Compare attributes of the stores so we can track
+			// ranges that have enough replicas but still need to be migrated
+			// onto nodes with the desired attributes.
+			if len(raftStatus.Progress) >= len(zoneConfig.ReplicaAttrs) {
+				replicatedRangeCount++
+			}
+
+			// If the range has the leader lease, then it's available.
+			if _, expired := rng.HasLeaderLease(timestamp); !expired {
+				availableRangeCount++
+			} else {
+				// If there is no leader lease, then as long as more than 50%
+				// of the replicas are current then it is available.
+				current := 0
+				for _, progress := range raftStatus.Progress {
```

Comment:
We need a more concrete solution for an availability metric.  It doesn't have to be perfect, but from looking at some of the other distributed db UIs, it's one of the most prominent metrics.  But as Ben mentioned, we can't just do this in a bottom up way, but we need more of a top down approach were we scan through all ranges.  
Here's Crate's admin ui:
![crate](https://cloud.githubusercontent.com/assets/1614265/7785236/bc0fa7c8-0154-11e5-8aff-a8ffab215012.PNG)